### PR TITLE
530 remove name from user

### PR DESCRIPTION
**Pull request by anncatton**

### CODE REVIEW

**Review by rtisma — CHANGES_REQUESTED** [review]
Great work, just some minor changes. In regards to the permission tests, that level of inheritance is not maintainable and generally discourage, however given the fact its the only way to create common test code (i.e common test code via composition is not supported), its acceptable for this special case.

File: src/main/java/bio/overture/ego/service/UserPermissionService.java [hunk 1]
```diff
@@ -52,6 +52,13 @@ public UserPermissionService(
     this.apiKeyEventsPublisher = apiKeyEventsPublisher;
   }
 
+  protected PolicyResponse convertToPolicyResponse(@NonNull UserPermission userPermission) {
```
Comment (rtisma) [inline src/main/java/bio/overture/ego/service/UserPermissionService.java:55]: add `@Override`

File: src/main/proto/Ego.proto [hunk 2]
```diff
@@ -59,4 +58,6 @@ message User {
 
     google.protobuf.StringValue provider_type = 14;
     google.protobuf.StringValue provider_id = 15;
+
+    reserved 7;
```
Comment (rtisma) [inline src/main/proto/Ego.proto:62]: add `reserved name` and also add a comment 
```
// remove the name = 7 field as part of EGO-### ticket
```

**Review by rtisma — CHANGES_REQUESTED** [review]

File: src/main/java/bio/overture/ego/service/UserPermissionService.java (hunk 1 above)
Comment (anncatton) [reply src/main/java/bio/overture/ego/service/UserPermissionService.java:55]: i removed the method from the AbstractPermissionService so there's nothing to override...AbstractNameablePermissionService has its own implementation that uses `owner.getName`

**Review by rtisma — APPROVED** [review]
Great work. Dont forget to update your "Summary" list of all the major changes  and breaking changes

**Pull request merged**